File: Hash-Comb/src/khalifa/ebtic/security/hashcomb/tree/transformer/Decoder.java
package khalifa.ebtic.security.hashcomb.tree.transformer;

import java.util.HashMap;
import java.util.Vector;

import khalifa.ebtic.security.hashcomb.tree.Node;
import khalifa.ebtic.security.hashcomb.tree.Tree;
import khalifa.ebtic.security.hashcomb.tree.Utils;

public class Decoder {
	


	private HashMap<String, Node> myMap;
	
	public Decoder()
	{
		this.myMap = Utils.readHashTable2File("configuration.ser");
	}
	
	
	public double decode(String encValue)
	{
		
		Node node = myMap.get(encValue);
		double center = node.getCenter();
		return center;
	}
	
	
}

File: Hash-Comb-Python/src/decoder.py
from __future__ import annotations
from typing import Optional, List, Any, Union, Dict, Sequence
import numpy as np

from src.node import Node
from src.tree import Tree
from src.utils import Utils

class Decoder:
    hashMap: Dict[str, Node]

    def __init__(self, configPath: str = "configuration.pkl") -> None:
        self.hashMap = Utils.readHashTable2File(configPath)

    def decode(self, encValue:str) -> float:
        node : Node = self.hashMap.get(encValue)
        if node is None: raise KeyError(f"hash {encValue} not found")
        return float(node.getCenter())

    def decodeArray(self, encValues: Union[Sequence[str], np.ndarray] ) -> np.ndarray:
        decodedStrings = np.empty(len(encValues), dtype=np.float64)
        for index, value in enumerate(encValues):
            decodedStrings[index] = self.decode(str(value))
        return decodedStrings
    